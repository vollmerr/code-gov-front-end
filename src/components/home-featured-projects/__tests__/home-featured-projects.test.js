import React from 'react';

import HomeFeaturedProjects from 'components/home-featured-projects'
import HomeFeaturedProject from 'components/home-featured-project'

import { mount, shallow } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk'

const middlewares = [thunk]
const mockStore = configureMockStore(middlewares);

describe('Testing <HomeFeaturedProjects /> component', () => {
  let wrapper;

  const props = {};

  const store = mockStore(props);

  beforeEach(() => {
    wrapper = mount(<HomeFeaturedProjects store={store} />);
  });
  it('Testing if component renders', () => {
    expect(wrapper.exists()).toEqual(true);
  });
  it('Testing if HomeFeaturedProject exists', () => {
    expect(wrapper.find(HomeFeaturedProject).exists()).toEqual(true);
  });
  it('should render correct', () => {
    expect(wrapper).toMatchSnapshot();
  })
})