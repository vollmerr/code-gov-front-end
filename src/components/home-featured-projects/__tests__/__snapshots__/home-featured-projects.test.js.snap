// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <HomeFeaturedProjects /> component should render correct 1`] = `
<Connect(HomeFeaturedProjects)
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <HomeFeaturedProjects
    dispatch={[Function]}
    featuredProjects={
      Array [
        Object {
          "alt": "test-alt",
          "description": "test-description",
          "image": "fa fa-comments",
          "links": Array [
            Object {
              "name": "test-link-name",
              "url": "https://test-link.com",
            },
          ],
          "short_name": "test-short-name",
        },
      ]
    }
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
  >
    <section
      className="featured-projects"
    >
      <HomeFeaturedProject
        index={0}
        key="test-short-name"
        project={
          Object {
            "alt": "test-alt",
            "description": "test-description",
            "image": "fa fa-comments",
            "links": Array [
              Object {
                "name": "test-link-name",
                "url": "https://test-link.com",
              },
            ],
            "short_name": "test-short-name",
          }
        }
      >
        <div
          className="block featured-project"
        >
          <div
            className="indented"
          >
            <hr />
            <div
              className="width-third"
            >
              <h2
                className="fp-short-name"
              >
                test-short-name
              </h2>
              <div
                align="center"
              >
                <i
                  alt="test-alt"
                  className="fa fa-comments"
                />
              </div>
            </div>
            <div
              className="width-two-thirds"
            >
              <p
                className="fp-description"
              >
                test-description
              </p>
              <span
                key="https://test-link.com"
              >
                <a
                  href="https://test-link.com"
                >
                  <button>
                    test-link-name
                  </button>
                </a>
              </span>
            </div>
          </div>
        </div>
      </HomeFeaturedProject>
    </section>
  </HomeFeaturedProjects>
</Connect(HomeFeaturedProjects)>
`;
